src/Application/Actions/Donations/Create.php

```diff
@@ -68,6 +68,27 @@ protected function action(): Response
 
         try {
             $donation = $this->donationRepository->buildFromApiRequest($donationData);
+
+            // Optionally set these donation fields if available
+            if (isset($donationData->billingPostalAddress)) {
+                $donation->setDonorPostalAddress((string) $donationData->billingPostalAddress);
```

Comment:
The values which are strings should not need casting to string. What is the use case here? We might also want to consider whether blank string is helpful and test strings for `empty()` instead if it's better to leave the fields null.